neuro_san/session/async_direct_agent_session.py

```diff
@@ -77,9 +77,9 @@ async def function(self, request_dict: Dict[str, Any]) -> Dict[str, Any]:
         response_dict: Dict[str, Any] = {
         }
 
-        front_man: str = self.tool_registry.find_front_man()
+        front_man: str = self.agent_network.find_front_man()
         if front_man is not None:
-            spec: Dict[str, Any] = self.tool_registry.get_agent_tool_spec(front_man)
+            spec: Dict[str, Any] = self.agent_network.get_agent_tool_spec(front_man)
```

Comment:
Interesting to note that most of the simple capabilities of the Sessions only need the AgentNetwork.
Yet another reason to leave the more complex functionality of config + factory to the DataDrivenChatSession class